.github/docker-node/16-centos7-build.Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM centos:7
+
+ENV NODE_VERSION 16.3.0
+
+# install software required for this OS
+RUN yum -y install \
+  python2 \
+  make
+
+# yum install gcc-c++  will install version 4.8.5
+# we need a higher version that supports c++ 14.
+# following will get version 7.3.1 and set as default
+RUN yum install -y centos-release-scl
+RUN yum install -y devtoolset-7-gcc*
```

Comment:
It seems devtoolset-8 is the recommended, which installs GCC 8.3 while devtoolset-7 installs GCC 7.3 -- should we just go with devtoolset-8 (or add a comment why if we don't)?

Reply:
I would vote for building directly with GCC 8.3, as this is the lowest required version for Node 16 as pointed out by @cheempz (https://github.com/nodejs/node/blob/master/BUILDING.md#supported-toolchains) and there does not seem much value in providing this sort pf "backwards compatibility" if it is not even supported by Node

Reply:
Agreed. Excellent link @cheempz. Was missed in the "quest to go back in time". 

Fixed. now using GCC 8.3.1 form `devtoolset-8`.